fn_grpc_interface/fn_grpc_interface/util/grpc_helper.py

```diff
@@ -0,0 +1,165 @@
+            elif six.PY2:
+                for file_name in interface_files:
+                    if file_name.endswith('.py'):
+                        module_name = file_name.split('.')[0]
+                        module_obj = imp.load_source(module_name, abs_file_path_json[file_name])
+                        interface_module.append(module_obj)
+            else:
+                raise ValueError("We do not support this version of python")
+
+            return interface_module
+        else:
+            self.log_object.debug("No Interface file found, please specify the interface file")
+            return None
+
+    def get_grpc_class(self, grpc_module_list, class_name):
+        """
+        component that return the given class object from the list of given module objects,
+        evaluates based on the given class names.
+        :param grpc_module_list: imported interface pb2 files module objects list
+        :param class_name:  interested class name in the given module objects list
+        :return:  tuple of name and class object of the given class name if class name found in the given module
+        objects, else returns an empty tuple.
+        """
+        for grpc_module in grpc_module_list:
+            for name, obj in inspect.getmembers(grpc_module, inspect.isclass):
+                self.log_object.debug("Class Name : {} Object : {}".format(name, obj))
+                if name.lower().find(class_name.lower()) != -1:
+                    return name, obj
+        else:
+            return ()
+
+    def gRPC_Connect_Server(self, auth_type=None, channel=None, stub_class=None, certificate_path=None):
+        """
+
+        :param auth_type: Input Secure Connection type
+        :param channel: gRPC Channel
+        :param stub_class: a Stub Class object from imported pb2 library
+        :param certificate_path: path of the ssl/tls certificate
+        :return: stub class object
+        """
+        # Checking supplied Input Arguments
+        if channel is None:
+            raise ValueError("channel argument should not be None type")
+        if stub_class is None:
+            raise ValueError("stub class object should not be None type")
+
+        if auth_type.lower().strip() not in ['ssl', 'tls', 'oauth2'] and auth_type.lower().strip() == "none":
+            self.log_object.info("gRPC Channel Connection is not secure.")
+            channel_object = grpc.insecure_channel(channel)
+            stub_class_obj = stub_class(channel_object)
+            return stub_class_obj, channel_object
+        elif auth_type.lower().strip().find('ssl') != -1 or auth_type.lower().strip().find('tls') != -1:
+            self.log_object.info("Channel Authentication is secure : {0}".format(auth_type))
+            if certificate_path is None:
+                raise ValueError("Please specify the valid certificate path")
+            if certificate_path.lower() == 'none':
```

Comment:
* Here `certificate_path` is taken from the app.config file and we are checking if it is `None` or `"none"
* This check should not be done here
* We should do the check when retrieve the variable from the app.config file and convert it to the correct Python Type
* This also goes for auth_type too

Reply:
ok